### ISSUE TITLE
number of times copied

### ISSUE THREAD

**Issue by lottybrand** [lottybrand22/lottysBartlett#12]
Eventually we'll need to display the number of times the non-copiers have previously been copied so far. So we probably want to assign something like property2 as a counter of times that player_id has been chosen by copiers so far in the game?

**Comment by thomasmorgan** [comment]
yes, this is sensible. If you look at the base class agent in dallinger/dallinger/nodes.py you can see how property1 is set up to behave like "fitness":

```
class Agent(Node):
    """A Node with fitness."""

    __mapper_args__ = {"polymorphic_identity": "agent"}

    @hybrid_property
    def fitness(self):
        """Endow agents with a numerical fitness."""
        try:
            return float(self.property1)
        except TypeError:
            return None

    @fitness.setter
    def fitness(self, fitness):
        """Assign fitness to property1."""
        self.property1 = repr(fitness)

    @fitness.expression
    def fitness(self):
        """Retrieve fitness via property1."""
        return cast(self.property1, Float)
```

**Comment by lottybrand** [comment]
Hi Tom, 

Thanks... although I'm not sure I follow what you mean. Do we need to assign property2 in Dallinger/dallinger/nodes.py  in a similar way to how property 1 is assigned here? (surely I don't have to change anything in the Dallinger code itself...?) 
I was under the impression we could just do something similar to line 71 in experiment.py but with node.property2 or would that not be sufficient?

**Comment by lottybrand** [comment]
So actually I just pushed my attempt at assigning something to property2 which seemed to work (I just copied a previous line but for property2). So now I just need to figure out how to count up the number of times each node was copied, right, and put that in place of line 72 in experiment.py ? (and then of course reference property2 in experiment.js to display this later).

**Comment by thomasmorgan** [comment]
The code I pasted (or something like it) can be included in your model.py if you define a custom type of node. It's not necessary, but it would allow you to do something like "node.copies" instead of "node.property2" and then have to remember that property2 is the number of copies.

But yes, what you have done is perfectly fine. What you'll want to do is have "info_post_request()" keep an eye one people's decisions and when someone decides to copy someone else it will need to add 1 to the value of their property2.

**Comment by lottybrand** [comment]
I've tried to add something to info_post_request() so that if the contents of an info matches property1 of that node (e.g. if A1 is selected to copy from) then it increments property2. So far property2 is showing up as zero, but it isn't incrementing because it freezes after the first copy (it doesn't increment for that copy anyway) because I think the positioning of my edit is stopping it getting past those lines. I've tried positioning it at the top and bottom of the info_post_request but it doesn't like either... I've pushed what I have so far..
